src/aws-cpp-sdk-core/source/client/ClientConfiguration.cpp

```diff
@@ -278,12 +280,21 @@ void setConfigFromEnvOrProfile(ClientConfiguration &config)
     if (disableIMDSv1 == "true") {
         config.disableImdsV1 = true;
     }
+
+    // accountId is intentionally not set here: AWS_ACCOUNT_ID env variable may not match the provided credentials.
+    // it must be set by an auth provider / identity resolver or by an SDK user.
+    config.accountIdEndpointMode = ClientConfiguration::LoadConfigFromEnvOrProfile(AWS_ACCOUNT_ID_ENDPOINT_MODE_ENVIRONMENT_VARIABLE,
+        config.profileName,
+        AWS_ACCOUNT_ID_ENDPOINT_MODE_CONFIG_FILE_OPTION,
+        {"required", "disabled", "preferred"}, /* allowed values */
+        "preferred" /* default value */);
 }
 
 ClientConfiguration::ClientConfiguration()
 {
     this->disableIMDS = false;
     setLegacyClientConfigurationParameters(*this);
+    setConfigFromEnvOrProfile(*this);;
```

Comment:
interesting fact -- this meant functionally that the `disableIMDSv1` configuration did not work if a user set a region in configuration